src/Quartz/Collection/TreeSet.cs

```diff
@@ -20,26 +20,37 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Runtime.Serialization;
 
 namespace Quartz.Collection
 {
     /// <summary>
     /// Only for backwards compatibility with serialization!
     /// </summary>
     /// <typeparam name="T"></typeparam>
+#if BINARY_SERIALIZATION
     [Serializable]
+#endif // BINARY_SERIALIZATION
+    [DataContract]  // TODO (NetCore Port): Confirm that data contract serialization works as expected here
```

Comment:
DataContract attributes should not be needed with Json.NET? These pollute the code base a bit.